libminifi/include/core/state/nodes/AgentInformation.h

```diff
@@ -425,77 +427,121 @@ class AgentStatus : public StateMonitorNode {
 
   std::vector<SerializedResponseNode> serialize() {
     std::vector<SerializedResponseNode> serialized;
+    if (!repositories_.empty()) {
+      serialized.push_back(serializeRepositories());
```

Comment:
after the refactor we should only depend on the return value of `serializeRepositories()` to determine if we should append it to the `serialized` vector, I propose either

1. an `SerializedResponseNode::empty` member function
2. or `serializeRepositories()` returning an `optional<...>` 

my preference lies with 1) az I wouldn't consider serializing an empty list of repositories "invalid" per-se 

(this applies to `serializeComponents` as well)

Reply:
good idea, went with your first suggestion in [d28d399](https://github.com/martinzink/nifi-minifi-cpp/commit/d28d3994fe633513158fb739edc49f69023b8a39)